HW2/P2/P2.txt

```diff
@@ -0,0 +1,41 @@
+# solution to P2
+
+For both uncorrelated or correlated problem, the serial version is very fast, much faster than any parallel version I could provide. It is because in this problem parallel version creates locks and derives overheads which take up most of the time.
```

Comment:
Yep, the task isn't compute-bound (it's memory-bound), so parallelizing it doesn't help because then there are just more memory accesses and those vastly outweigh the compute contribution.